### PR TITLE
smart contracts clarifications

### PR DESCRIPTION
**Pull request by nadinequinr3**

### CODE REVIEW

**Review by paulmoloneyr3 — APPROVED** [review]

File: content/en/platform/corda/5.0-beta/developing/ledger/smart-contracts.md [hunk 1]
```diff
@@ -38,14 +37,14 @@ Each transaction state specifies a contract type. The contract specified takes t
 The contract code can:
 
 * Check the number of inputs, outputs, commands, or attachments.
-* Check for time windows.
+* Check that a [time window]({{< relref "./time-windows.md" >}}) exists and that time points mentioned in states fall within this time window.
```
[the remark this answers is not in the document]
Comment (colin-hillis) [reply content/en/platform/corda/5.0-beta/developing/ledger/smart-contracts.md:40]: ^^^ assuming that 'States' is another field or function ^^^

**Review by colin-hillis — CHANGES_REQUESTED** [review]
Couple of small changes. Take or leave the first one.

File: content/en/platform/corda/5.0-beta/developing/ledger/smart-contracts.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (blsemo) [reply content/en/platform/corda/5.0-beta/developing/ledger/smart-contracts.md:40]: I am not sure I would word it that specifically - that was just the most obvious example. The contract verification code can check that the time window matches the expectations for the given transaction, e.g. that a point in time mentioned in one of the states (inputs, outputs, references) falls within the time window.
Also, a time window always exists, as in Corda 5, it's mandatory for all UTXO transactions. You wouldn't even get this far without one.

File: content/en/platform/corda/5.0-beta/developing/ledger/smart-contracts.md [hunk 2]
```diff
@@ -38,14 +37,14 @@ Each transaction state specifies a contract type. The contract specified takes t
...
 * Check the contents of all components.
 * Evaluate looping constructs, variable assignments, function calls, helper methods, and other aspects of the transaction code.
 * Group similar states to validate them as a group. For example, it can impose a rule on the combined value of all the cash states.
 
 ## Determinism
 
-For the nodes on a network to reach consensus about a proposed update to the ledger, transaction verification must be deterministic. That means contracts must always accept or always reject a given transaction. For example, a transaction’s validity cannot depend on the time it was validated, or the amount of information the node running the contract holds.
+To enable the nodes on a network to reach consensus about a proposed update to the ledger, transaction verification must be deterministic. This means that contracts must always accept or reject a given transaction, regardless of when it is validated, or the amount of information that the node running the contract holds.
```
Comment (blsemo) [inline content/en/platform/corda/5.0-beta/developing/ledger/smart-contracts.md:47]: Hmm - that last sentence is a bit misleading, it's always a reject or an accept, the point being that the result can't change on further validations once it's accepted or rejected.

**Review by blsemo — CHANGES_REQUESTED** [review]
While not wrong, I am not 100% convinced this is worded the most understandable way.

**Comment by nadinequinr3** [comment]
@blsemo I've made some updates, if you could re-review, please.

**Review by yarivus — APPROVED** [review]
LGTM

**Review by yarivus — APPROVED** [review]
LGTM - clean merge

**Review by blsemo — APPROVED** [review]
LGTM

**Review by yarivus — APPROVED** [review]
LGTM - clean merge

**Review by yarivus — APPROVED** [review]
LGTM - clean merge

**Review by colin-hillis — APPROVED** [review]
LGTM

**Pull request merged**